msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/ltw/TestCase2582292.java

```diff
@@ -0,0 +1,150 @@
+import com.microsoft.identity.client.ui.automation.annotations.LTWTests;
+import com.microsoft.identity.client.ui.automation.annotations.SupportedBrokers;
+import com.microsoft.identity.client.ui.automation.app.MsalTestApp;
+import com.microsoft.identity.client.ui.automation.broker.BrokerLTW;
+import com.microsoft.identity.client.ui.automation.broker.BrokerMicrosoftAuthenticator;
+import com.microsoft.identity.client.ui.automation.interaction.PromptParameter;
+import com.microsoft.identity.client.ui.automation.interaction.microsoftsts.MicrosoftStsPromptHandlerParameters;
+import com.microsoft.identity.labapi.utilities.client.ILabAccount;
+import com.microsoft.identity.labapi.utilities.client.LabQuery;
+import com.microsoft.identity.labapi.utilities.constants.AzureEnvironment;
+import com.microsoft.identity.labapi.utilities.constants.TempUserType;
+import com.microsoft.identity.labapi.utilities.constants.UserRole;
+import com.microsoft.identity.labapi.utilities.constants.UserType;
+import com.microsoft.identity.labapi.utilities.exception.LabApiException;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.List;
+
+// If LTW is the active broker, and request is made through Authenticator from an old MSAL in shared device mode, nothing should break
+// https://identitydivision.visualstudio.com/Engineering/_workitems/edit/2582292
+@LTWTests
+@SupportedBrokers(brokers = {BrokerLTW.class})
+public class TestCase2582292 extends AbstractMsalBrokerTest {
+
+    @Test
+    public void test_2582292() throws LabApiException, InterruptedException, UiObjectNotFoundException {
+        final String username = mLabAccount.getUsername();
+        final String password = mLabAccount.getPassword();
+
+        // Install new Auth app with broker SDK changes of broker selection logic
+        final BrokerMicrosoftAuthenticator brokerMicrosoftAuthenticator = new BrokerMicrosoftAuthenticator();
+        brokerMicrosoftAuthenticator.install();
+
+        // Install legacy MSAL Test app (Msal test app with no broker selection logic)
+        final MsalTestApp msalTestApp = new MsalTestApp();
+        msalTestApp.installOldApk();
+
+        // Set up shared device mode
+        // Open Authenticator app -> ... -> Settings
+        brokerMicrosoftAuthenticator.performSharedDeviceRegistration(username, password);
+
+        // Check mode in MSAL test app
+        msalTestApp.launch();
+        msalTestApp.handleFirstRun();
+        final String mode = msalTestApp.checkMode();
+        Assert.assertTrue(mode.contains("Single Account"));
+
+        // performs AcquireToken with an account from the same tenant with the WPJed account.
+        final LabQuery query = LabQuery.builder()
+                .userType(UserType.CLOUD)
+                .build();
+
+        final ILabAccount difAccount = mLabClient.getLabAccount(query);
+        final String usernameDif = difAccount.getUsername();
+        final String passwordDif = difAccount.getPassword();
```

Comment:
variable names as username2 and password2?

Reply:
Sure, I've changed them to username2 and password2. I also changed username and password to username1 and password1 for consistency.